### PR TITLE
Implement operations of atomics

### PR DESCRIPTION
**Pull request by gergocs**
The following methods were implemented:
Atomics.compareExchange
Atomics.store

The implementation is based on PR #4956, only resolved the conflicts.

Co-authored-by: Csaba Repasi repasics@inf.u-szeged.hu
JerryScript-DCO-1.0-Signed-off-by: Gergo Csizi gergocs@inf.u-szeged.hu

### CODE REVIEW

**Review by LaszloLango — CHANGES_REQUESTED** [review]
Could you please add more details into the commit message, what operations are implemented and what are still missing?

**Comment by ossy-szeged** [comment]
Please add proper ifdef guard not to break non bigint build. See #5047 for details.

**Review by zherczeg — APPROVED** [review]
LGTM. Nice patch!

**Review by LaszloLango — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/operations/ecma-arraybuffer-object.c [hunk 1]
```diff
@@ -553,6 +556,163 @@ ecma_builtin_arraybuffer_slice (ecma_value_t this_arg, const ecma_value_t *argum
...
+ * @param indexed_position: position of the element in buffer
+ * @param val: parameter of op, number or bigint
+ * @param op: the atomics operation to execute
+ * @param element_type: type of the elements stored in typedArray
+ * @param typedarray_getter_cb: getter callback of typedArray
+ * @param typedarray_setter_cb: setter callback of typedArray
+ *
+ * @return ecma value
+ */
+ecma_value_t
```
[the remark this answers is not in the document]
Comment (gergocs) [reply jerry-core/ecma/operations/ecma-arraybuffer-object.c:667]: It is used in an other file so I don't think it should be static.

**Review by LaszloLango — APPROVED** [review]
LGTM

**Pull request merged**